front/app/utils/actionDescriptors/index.ts

```diff
@@ -29,27 +20,54 @@ export const isFixableByAuthentication = (disabledReason: string) => {
   return FIXABLE_REASONS.has(disabledReason);
 };
 
-export type ActionDescriptor<DisabledReason> =
-  | { enabled: true; disabled_reason: null }
-  | { enabled: false; disabled_reason: DisabledReason };
-
-// If future_enabled_at is a string, it's a date
-export type ActionDescriptorFutureEnabled<DisabledReason> =
-  | { enabled: true; disabled_reason: null; future_enabled_at: null }
-  | {
-      enabled: false;
-      disabled_reason: DisabledReason;
-      future_enabled_at: string | null;
-    };
-
-// NOTE: Bit of a shim to add in the budgeting action - even though it doesn't really exist
-type ActionDescriptorActions = 'voting' | 'budgeting';
+// Fall back messages for disabled reasons
+const globalDisabledMessages: {
+  [reason in DisabledReason]?: MessageDescriptor;
+} = {
+  user_not_in_group: messages.defaultNotInGroup,
+};
 
-const disabledMessages: {
-  [action in ActionDescriptorActions]?: {
-    [reason in UserDisabledReason]?: MessageDescriptor;
+// Messages specific to each action
+const actionDisabledMessages: {
```

Comment:
Is this a complete overview of all possible disabled reasons per action? It would be great if we could define this once and then derive all types from it. 

With a bit of restructuring we could also make it more clever, e.g. have a type error when you try to pick a non-existent disabled reason- e.g. when doing something like `getPermissionsDisabledMessage('taking_poll', 'reacting_disabled')`

Reply:
Can't say if it's a complete overview - but it should be a complete overview of all disabled actions that relate to the permissions on the backend. I'll let you add the extra ticket. Cheers